3rd_party/oxygine-framework/oxygine/core/opengl/UberShaderProgram.cpp

```diff
@@ -14,80 +14,28 @@ namespace oxygine
         releaseShaders();
     }
 
-    void UberShaderProgram::init()
-    {
-        QString fracShaderBody;
-        QString fracTableShaderBody;
-        QString fracMatrixShaderBody;
-        QString vertexShaderBody;
-        QString filepath = "system/frac_shader.glsl";
-        if (!QFile::exists(filepath))
-        {
-            filepath = oxygine::Resource::RCC_PREFIX_PATH + filepath;
-        }
+    static QString loadShader(QString name) {
+        QString filepath = oxygine::Resource::RCC_PREFIX_PATH + name;
         if (QFile::exists(filepath))
         {
             QFile file(filepath);
             file.open(QIODevice::ReadOnly);
             QTextStream stream(&file);
-            fracShaderBody = stream.readAll();
+            return stream.readAll();
         }
         else
         {
-            CONSOLE_PRINT("Unable to find frac shader: " + filepath, GameConsole::eERROR);
-        }
-        filepath = "system/vertex_shader.glsl";
-        if (!QFile::exists(filepath))
-        {
-            filepath = oxygine::Resource::RCC_PREFIX_PATH + filepath;
+            CONSOLE_PRINT("Unable to find frac shader: " + filepath, GameConsole::eFATAL);
```

Comment:
Please remove frac from the message since it's a generic function now.
Please change the log level back to error.

Reply:
How would the game continue without the core shaders?

Reply:
Yes in certain cases the shaders aren't needed mainly when running without ui where the ui window still exists but is never shown. -> For the user of the game not relevant for me as a developer it's.